diff --git a/makepanda/makepandacore.py b/makepanda/makepandacore.py
index ad75ae6f1d..38a7f16a3c 100644
--- a/makepanda/makepandacore.py
+++ b/makepanda/makepandacore.py
@@ -2176,6 +2176,14 @@ def SdkLocatePython(prefer_thirdparty_python=False):
     abiflags = getattr(sys, 'abiflags', '')
 
     if GetTarget() == 'windows':
+        SDK["PYTHONVERSION"] = "python3.11"
+        SDK["PYTHONEXEC"] = os.path.realpath(sys.executable)
+        SDK["PYTHON"] = os.path.dirname(os.path.dirname(SDK["PYTHONEXEC"]))
+
+        IncDirectory("PYTHON", SDK["PYTHON"] + "/include")
+        LibDirectory("PYTHON", SDK["PYTHON"] + "/lib")
+
+    elif GetTarget() == 'not_windows':
         sdkdir = GetThirdpartyBase() + "/win-python"
 
         if sys.version_info >= (3, 0):
