diff --git a/makepanda/makepandacore.py b/makepanda/makepandacore.py
index 55cb8dfa2f..77a21f86a0 100644
--- a/makepanda/makepandacore.py
+++ b/makepanda/makepandacore.py
@@ -2155,7 +2155,7 @@ def SdkLocatePython(prefer_thirdparty_python=False):
 
         sdkdir = sdkdir.replace('\\', '/')
         SDK["PYTHON"] = sdkdir
-        SDK["PYTHONEXEC"] = SDK["PYTHON"] + "/python"
+        SDK["PYTHONEXEC"] = os.path.realpath(sys.executable).rstrip('.exe')
 
         if (GetOptimize() <= 2):
             SDK["PYTHONEXEC"] += "_d.exe"
@@ -2171,7 +2171,13 @@ def SdkLocatePython(prefer_thirdparty_python=False):
                       glob.glob(SDK["PYTHON"] + "/python[0-9][0-9][0-9]_d.dll")
         else:
             py_dlls = glob.glob(SDK["PYTHON"] + "/python[0-9][0-9].dll") + \
-                      glob.glob(SDK["PYTHON"] + "/python[0-9][0-9][0-9].dll")
+                      glob.glob(SDK["PYTHON"] + "/python[0-9][0-9][0-9].dll") + \
+                      glob.glob(SDK["PYTHON"] + "/python[0-9][0-9]t.dll") + \
+                      glob.glob(SDK["PYTHON"] + "/python[0-9][0-9][0-9]t.dll") + \
+                      glob.glob(SDK["PYTHON"] + "/bin/python[0-9][0-9].dll") + \
+                      glob.glob(SDK["PYTHON"] + "/bin/python[0-9][0-9][0-9].dll") + \
+                      glob.glob(SDK["PYTHON"] + "/bin/python[0-9][0-9]t.dll") + \
+                      glob.glob(SDK["PYTHON"] + "/bin/python[0-9][0-9][0-9]t.dll")
 
         if len(py_dlls) == 0:
             exit("Could not find the Python dll in %s." % (SDK["PYTHON"]))
@@ -2182,7 +2188,7 @@ def SdkLocatePython(prefer_thirdparty_python=False):
         py_dllver = py_dll.strip(".DHLNOPTY_dhlnopty")
         ver = py_dllver[0] + '.' + py_dllver[1:]
 
-        SDK["PYTHONVERSION"] = "python" + ver
+        SDK["PYTHONVERSION"] = py_dll.rpartition('/')[-1].rstrip('.dll')
         os.environ["PYTHONHOME"] = SDK["PYTHON"]
 
         running_ver = '%d.%d' % sys.version_info[:2]
