--- build_files/cmake/platform/platform_win32_msvc.cmake.orig	2019-02-22 15:53:50.362271425 +0300
+++ build_files/cmake/platform/platform_win32_msvc.cmake	2019-02-23 00:02:59.639404503 +0300
@@ -186,7 +186,7 @@
 endif()
 
 set(PTHREADS_INCLUDE_DIRS ${LIBDIR}/pthreads/include)
-set(PTHREADS_LIBRARIES ${LIBDIR}/pthreads/lib/pthreadVC2.lib)
+set(PTHREADS_LIBRARIES ${LIBDIR}/pthreads/lib/pthreadVC3.lib)
 
 set(FREETYPE ${LIBDIR}/freetype)
 set(FREETYPE_INCLUDE_DIRS
@@ -266,16 +266,16 @@
 		set(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR} ${OPENEXR}/include/OpenEXR)
 		set(OPENEXR_LIBPATH ${OPENEXR}/lib)
 		set(OPENEXR_LIBRARIES
-			optimized ${OPENEXR_LIBPATH}/Iex-2_2.lib
-			optimized ${OPENEXR_LIBPATH}/Half.lib
-			optimized ${OPENEXR_LIBPATH}/IlmImf-2_2.lib
-			optimized ${OPENEXR_LIBPATH}/Imath-2_2.lib
-			optimized ${OPENEXR_LIBPATH}/IlmThread-2_2.lib
-			debug ${OPENEXR_LIBPATH}/Iex-2_2_d.lib
-			debug ${OPENEXR_LIBPATH}/Half_d.lib
-			debug ${OPENEXR_LIBPATH}/IlmImf-2_2_d.lib
-			debug ${OPENEXR_LIBPATH}/Imath-2_2_d.lib
-			debug ${OPENEXR_LIBPATH}/IlmThread-2_2_d.lib
+			optimized ${OPENEXR_LIBPATH}/Iex_s.lib
+			optimized ${OPENEXR_LIBPATH}/Half_s.lib
+			optimized ${OPENEXR_LIBPATH}/IlmImf_s.lib
+			optimized ${OPENEXR_LIBPATH}/Imath_s.lib
+			optimized ${OPENEXR_LIBPATH}/IlmThread_s.lib
+			debug ${OPENEXR_LIBPATH}/Iex_s_d.lib
+			debug ${OPENEXR_LIBPATH}/Half_s_d.lib
+			debug ${OPENEXR_LIBPATH}/IlmImf_s_d.lib
+			debug ${OPENEXR_LIBPATH}/Imath_s_d.lib
+			debug ${OPENEXR_LIBPATH}/IlmThread_s_d.lib
 		)
 	endif()
 endif()
