diff --git a/build_files/cmake/platform/platform_win32.cmake b/build_files/cmake/platform/platform_win32.cmake
index 827357df043..3cb2e3f2318 100644
--- a/build_files/cmake/platform/platform_win32.cmake
+++ b/build_files/cmake/platform/platform_win32.cmake
@@ -334,16 +334,6 @@ if(WITH_JACK)
 endif()
 
 if(WITH_PYTHON)
-	set(PYTHON_VERSION 3.7) # CACHE STRING)
-
-	string(REPLACE "." "" _PYTHON_VERSION_NO_DOTS ${PYTHON_VERSION})
-	# Use shared libs for vc2008 and vc2010 until we actually have vc2010 libs
-	set(PYTHON_LIBRARY ${LIBDIR}/python/lib/python${_PYTHON_VERSION_NO_DOTS}.lib)
-	unset(_PYTHON_VERSION_NO_DOTS)
-
-	# Shared includes for both vc2008 and vc2010
-	set(PYTHON_INCLUDE_DIR ${LIBDIR}/python/include/python${PYTHON_VERSION})
-
 	# uncached vars
 	set(PYTHON_INCLUDE_DIRS "${PYTHON_INCLUDE_DIR}")
 	set(PYTHON_LIBRARIES  "${PYTHON_LIBRARY}")
