--- source/creator/CMakeLists.txt.orig	2018-03-23 18:22:25.000000000 +0300
+++ source/creator/CMakeLists.txt	2019-02-22 15:50:37.536024902 +0300
@@ -641,20 +641,6 @@
 	if(WITH_PYTHON)
 		string(REPLACE "." "" _PYTHON_VERSION_NO_DOTS ${PYTHON_VERSION})
 
-		if(NOT CMAKE_COMPILER_IS_GNUCC)
-			install(
-				FILES ${LIBDIR}/python/lib/python${_PYTHON_VERSION_NO_DOTS}.dll
-				DESTINATION "."
-				CONFIGURATIONS Release;RelWithDebInfo;MinSizeRel
-			)
-
-			install(
-				FILES ${LIBDIR}/python/lib/python${_PYTHON_VERSION_NO_DOTS}_d.dll
-				DESTINATION "."
-				CONFIGURATIONS Debug
-			)
-		endif()
-
 		if(WITH_PYTHON_INSTALL)
 			# note, as far as python is concerned 'RelWithDebInfo' is not debug since its without debug flags.
 
