diff --git a/build_files/cmake/platform/platform_win32.cmake b/build_files/cmake/platform/platform_win32.cmake
index 7a2d3ad948a..a8fb697d66d 100644
--- a/build_files/cmake/platform/platform_win32.cmake
+++ b/build_files/cmake/platform/platform_win32.cmake
@@ -408,8 +408,8 @@ if(WITH_IMAGE_OPENEXR)
     set(IMATH_INCLUDE_DIRS ${IMATH_INCLUDE_DIR} ${IMATH}/include/Imath)
     set(IMATH_LIBPATH ${IMATH}/lib)
     set(IMATH_LIBRARIES
-      optimized ${IMATH_LIBPATH}/Imath_s.lib
-      debug ${IMATH_LIBPATH}/Imath_s_d.lib
+      optimized ${IMATH_LIBPATH}/Imath.lib
+      debug ${IMATH_LIBPATH}/Imath_d.lib
     )
   endif()
   set(OPENEXR_ROOT_DIR ${LIBDIR}/openexr)
@@ -423,32 +423,32 @@ if(WITH_IMAGE_OPENEXR)
     set(OPENEXR_LIBPATH ${OPENEXR}/lib)
     # Check if the 3.x library name exists
     # if not assume this is a 2.x library folder
-    if(EXISTS "${OPENEXR_LIBPATH}/OpenEXR_s.lib")
+    if(EXISTS "${OPENEXR_LIBPATH}/OpenEXR.lib")
       set(OPENEXR_LIBRARIES
-        optimized ${OPENEXR_LIBPATH}/Iex_s.lib
-        optimized ${OPENEXR_LIBPATH}/IlmThread_s.lib
-        optimized ${OPENEXR_LIBPATH}/OpenEXR_s.lib
-        optimized ${OPENEXR_LIBPATH}/OpenEXRCore_s.lib
-        optimized ${OPENEXR_LIBPATH}/OpenEXRUtil_s.lib
-        debug ${OPENEXR_LIBPATH}/Iex_s_d.lib
-        debug ${OPENEXR_LIBPATH}/IlmThread_s_d.lib
-        debug ${OPENEXR_LIBPATH}/OpenEXR_s_d.lib
-        debug ${OPENEXR_LIBPATH}/OpenEXRCore_s_d.lib
-        debug ${OPENEXR_LIBPATH}/OpenEXRUtil_s_d.lib
+        optimized ${OPENEXR_LIBPATH}/Iex.lib
+        optimized ${OPENEXR_LIBPATH}/IlmThread.lib
+        optimized ${OPENEXR_LIBPATH}/OpenEXR.lib
+        optimized ${OPENEXR_LIBPATH}/OpenEXRCore.lib
+        optimized ${OPENEXR_LIBPATH}/OpenEXRUtil.lib
+        debug ${OPENEXR_LIBPATH}/Iex_d.lib
+        debug ${OPENEXR_LIBPATH}/IlmThread_d.lib
+        debug ${OPENEXR_LIBPATH}/OpenEXR_d.lib
+        debug ${OPENEXR_LIBPATH}/OpenEXRCore_d.lib
+        debug ${OPENEXR_LIBPATH}/OpenEXRUtil_d.lib
         ${IMATH_LIBRARIES}
       )
     else()
       set(OPENEXR_LIBRARIES
-        optimized ${OPENEXR_LIBPATH}/Iex_s.lib
-        optimized ${OPENEXR_LIBPATH}/Half_s.lib
-        optimized ${OPENEXR_LIBPATH}/IlmImf_s.lib
-        optimized ${OPENEXR_LIBPATH}/Imath_s.lib
-        optimized ${OPENEXR_LIBPATH}/IlmThread_s.lib
-        debug ${OPENEXR_LIBPATH}/Iex_s_d.lib
-        debug ${OPENEXR_LIBPATH}/Half_s_d.lib
-        debug ${OPENEXR_LIBPATH}/IlmImf_s_d.lib
-        debug ${OPENEXR_LIBPATH}/Imath_s_d.lib
-        debug ${OPENEXR_LIBPATH}/IlmThread_s_d.lib
+        optimized ${OPENEXR_LIBPATH}/Iex.lib
+        optimized ${OPENEXR_LIBPATH}/Half.lib
+        optimized ${OPENEXR_LIBPATH}/IlmImf.lib
+        optimized ${OPENEXR_LIBPATH}/Imath.lib
+        optimized ${OPENEXR_LIBPATH}/IlmThread.lib
+        debug ${OPENEXR_LIBPATH}/Iex_d.lib
+        debug ${OPENEXR_LIBPATH}/Half_d.lib
+        debug ${OPENEXR_LIBPATH}/IlmImf_d.lib
+        debug ${OPENEXR_LIBPATH}/Imath_d.lib
+        debug ${OPENEXR_LIBPATH}/IlmThread_d.lib
       )
     endif()
   endif()
@@ -473,20 +473,6 @@ if(WITH_JACK)
 endif()
 
 if(WITH_PYTHON)
-  # Cache version for make_bpy_wheel.py to detect.
-  unset(PYTHON_VERSION CACHE)
-  set(PYTHON_VERSION "3.10" CACHE STRING "Python version")
-
-  string(REPLACE "." "" _PYTHON_VERSION_NO_DOTS ${PYTHON_VERSION})
-  set(PYTHON_LIBRARY ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/libs/python${_PYTHON_VERSION_NO_DOTS}.lib)
-  set(PYTHON_LIBRARY_DEBUG ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/libs/python${_PYTHON_VERSION_NO_DOTS}_d.lib)
-
-  set(PYTHON_EXECUTABLE ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/bin/python$<$<CONFIG:Debug>:_d>.exe)
-
-  set(PYTHON_INCLUDE_DIR ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/include)
-  set(PYTHON_NUMPY_INCLUDE_DIRS ${LIBDIR}/python/${_PYTHON_VERSION_NO_DOTS}/lib/site-packages/numpy/core/include)
-  set(NUMPY_FOUND ON)
-  unset(_PYTHON_VERSION_NO_DOTS)
   # uncached vars
   set(PYTHON_INCLUDE_DIRS "${PYTHON_INCLUDE_DIR}")
   set(PYTHON_LIBRARIES debug "${PYTHON_LIBRARY_DEBUG}" optimized "${PYTHON_LIBRARY}" )
