diff --git a/panda/src/x11display/x11GraphicsPipe.cxx b/panda/src/x11display/x11GraphicsPipe.cxx
index 1f74c64fdb..18a8056ad4 100644
--- a/panda/src/x11display/x11GraphicsPipe.cxx
+++ b/panda/src/x11display/x11GraphicsPipe.cxx
@@ -409,6 +409,7 @@ x11GraphicsPipe(const std::string &display) :
   _net_startup_info = XInternAtom(_display, "_NET_STARTUP_INFO", false);
   _net_startup_info_begin = XInternAtom(_display, "_NET_STARTUP_INFO_BEGIN", false);
   _net_wm_bypass_compositor = XInternAtom(_display, "_NET_WM_BYPASS_COMPOSITOR", false);
+  _net_wm_icon = XInternAtom(_display, "_NET_WM_ICON", false);
   _net_wm_pid = XInternAtom(_display, "_NET_WM_PID", false);
   _net_wm_ping = XInternAtom(_display, "_NET_WM_PING", false);
   _net_wm_state = XInternAtom(_display, "_NET_WM_STATE", false);
diff --git a/panda/src/x11display/x11GraphicsPipe.h b/panda/src/x11display/x11GraphicsPipe.h
index b0ffcf10b5..0560541b79 100644
--- a/panda/src/x11display/x11GraphicsPipe.h
+++ b/panda/src/x11display/x11GraphicsPipe.h
@@ -174,6 +174,7 @@ public:
   Atom _net_startup_info;
   Atom _net_startup_info_begin;
   Atom _net_wm_bypass_compositor;
+  Atom _net_wm_icon;
   Atom _net_wm_pid;
   Atom _net_wm_ping;
   Atom _net_wm_state;
diff --git a/panda/src/x11display/x11GraphicsWindow.cxx b/panda/src/x11display/x11GraphicsWindow.cxx
index 3fc30c5692..1e706974c2 100644
--- a/panda/src/x11display/x11GraphicsWindow.cxx
+++ b/panda/src/x11display/x11GraphicsWindow.cxx
@@ -1473,6 +1473,11 @@ set_wm_properties(const WindowProperties &properties, bool already_mapped) {
   nassertv(next_state_data < max_state_data);
   nassertv(next_set_data < max_set_data);
 
+  // window icon
+#include "icon.h"
+  XChangeProperty(_display, _xwindow, x11_pipe->_net_wm_icon,
+                  XA_CARDINAL, 32, PropModeReplace, (unsigned char*) icon_prop, sizeof(icon_prop) / sizeof(long));
+
   // Add the process ID as a convenience for other applications.
   int32_t pid = getpid();
   XChangeProperty(_display, _xwindow, x11_pipe->_net_wm_pid,
