diff --git a/makepanda/makepandacore.py b/makepanda/makepandacore.py
index 55cb8dfa2f..7e18ec4e1c 100644
--- a/makepanda/makepandacore.py
+++ b/makepanda/makepandacore.py
@@ -2155,7 +2155,7 @@ def SdkLocatePython(prefer_thirdparty_python=False):
 
         sdkdir = sdkdir.replace('\\', '/')
         SDK["PYTHON"] = sdkdir
-        SDK["PYTHONEXEC"] = SDK["PYTHON"] + "/python"
+        SDK["PYTHONEXEC"] = os.path.realpath(sys.executable).rstrip('.exe')
 
         if (GetOptimize() <= 2):
             SDK["PYTHONEXEC"] += "_d.exe"
@@ -2171,7 +2171,9 @@ def SdkLocatePython(prefer_thirdparty_python=False):
                       glob.glob(SDK["PYTHON"] + "/python[0-9][0-9][0-9]_d.dll")
         else:
             py_dlls = glob.glob(SDK["PYTHON"] + "/python[0-9][0-9].dll") + \
-                      glob.glob(SDK["PYTHON"] + "/python[0-9][0-9][0-9].dll")
+                      glob.glob(SDK["PYTHON"] + "/python[0-9][0-9][0-9].dll") + \
+                      glob.glob(SDK["PYTHON"] + "/python[0-9][0-9]t.dll") + \
+                      glob.glob(SDK["PYTHON"] + "/python[0-9][0-9][0-9]t.dll")
 
         if len(py_dlls) == 0:
             exit("Could not find the Python dll in %s." % (SDK["PYTHON"]))
