diff --git a/panda/src/gobj/texture.cxx b/panda/src/gobj/texture.cxx
index 99c5d10c02..914649f626 100644
--- a/panda/src/gobj/texture.cxx
+++ b/panda/src/gobj/texture.cxx
@@ -2037,7 +2037,7 @@ down_to_power_2(int value) {
  */
 void Texture::
 consider_rescale(PNMImage &pnmimage) {
-  consider_rescale(pnmimage, get_name(), get_auto_texture_scale());
+  consider_rescale(pnmimage, get_fullpath(), get_auto_texture_scale());
 }
 
 /**
@@ -3160,7 +3160,7 @@ do_read_one(CData *cdata, const Filename &fullpath, const Filename &alpha_fullpa
       y_size = 1;
 
     } else {
-      adjust_size(x_size, y_size, fullpath.get_basename(), false, auto_texture_scale);
+      adjust_size(x_size, y_size, fullpath.get_fullpath(), false, auto_texture_scale);
     }
 
     if (read_floating_point) {
@@ -3184,7 +3184,7 @@ do_read_one(CData *cdata, const Filename &fullpath, const Filename &alpha_fullpa
       cdata->_orig_file_x_size = x_size;
       cdata->_orig_file_y_size = y_size;
 
-      if (adjust_size(x_size, y_size, fullpath.get_basename(), false, auto_texture_scale)) {
+      if (adjust_size(x_size, y_size, fullpath.get_fullpath(), false, auto_texture_scale)) {
         image.set_read_size(x_size, y_size);
       }
     } else {
@@ -3360,7 +3360,7 @@ do_read_one(CData *cdata, const Filename &fullpath, const Filename &alpha_fullpa
     if (do_get_auto_texture_scale(cdata) == ATS_pad) {
       int new_x_size = image.get_x_size();
       int new_y_size = image.get_y_size();
-      if (do_adjust_this_size(cdata, new_x_size, new_y_size, fullpath.get_basename(), true)) {
+      if (do_adjust_this_size(cdata, new_x_size, new_y_size, fullpath.get_fullpath(), true)) {
         pad_x_size = new_x_size - image.get_x_size();
         pad_y_size = new_y_size - image.get_y_size();
         PNMImage new_image(new_x_size, new_y_size, image.get_num_channels(),
@@ -5491,7 +5491,7 @@ do_reload_ram_image(CData *cdata, bool allow_compression) {
       // and we want a different-sized texture now.
       int x_size = cdata->_orig_file_x_size;
       int y_size = cdata->_orig_file_y_size;
-      do_adjust_this_size(cdata, x_size, y_size, cdata->_filename.get_basename(), true);
+      do_adjust_this_size(cdata, x_size, y_size, cdata->_filename.get_fullpath(), true);
       if (x_size != tex->get_x_size() || y_size != tex->get_y_size()) {
         if (gobj_cat.is_debug()) {
           gobj_cat.debug()
@@ -6812,7 +6812,7 @@ do_rescale_texture(CData *cdata) {
     return false;
   }
 
-  if (do_adjust_this_size(cdata, new_x_size, new_y_size, get_name(), false)) {
+  if (do_adjust_this_size(cdata, new_x_size, new_y_size, get_fullpath(), false)) {
     // OK, we have to scale the image.
     PNMImage orig_image;
     if (!do_store_one(cdata, orig_image, 0, 0)) {
@@ -6846,7 +6846,7 @@ do_rescale_texture(CData *cdata) {
   if (do_get_auto_texture_scale(cdata) == ATS_pad) {
     new_x_size = cdata->_x_size;
     new_y_size = cdata->_y_size;
-    if (do_adjust_this_size(cdata, new_x_size, new_y_size, get_name(), true)) {
+    if (do_adjust_this_size(cdata, new_x_size, new_y_size, get_fullpath(), true)) {
       pad_x_size = new_x_size - cdata->_x_size;
       pad_y_size = new_y_size - cdata->_y_size;
 
