diff --git a/makepanda/makepanda.py b/makepanda/makepanda.py
index 23fd57afaa..2c273cb4a4 100755
--- a/makepanda/makepanda.py
+++ b/makepanda/makepanda.py
@@ -976,6 +976,11 @@ if (COMPILER == "MSVC"):
         LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletCollision" + suffix)
         LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletDynamics" + suffix)
         LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletSoftBody" + suffix)
+        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3Common" + suffix)
+        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3Collision" + suffix)
+        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3Dynamics" + suffix)
+        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3Geometry" + suffix)
+        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3OpenCL_clew" + suffix)
 
 if (COMPILER=="GCC"):
     PkgDisable("MIMALLOC") # no discernable benefit over glibc
@@ -1051,7 +1056,7 @@ if (COMPILER=="GCC"):
         SmartPkgEnable("TIFF",      "libtiff-4", ("tiff"), "tiff.h")
         SmartPkgEnable("OPENEXR",   "OpenEXR",   ("IlmImf", "Imath", "Half", "Iex", "IexMath", "IlmThread"), ("OpenEXR", "Imath", "OpenEXR/ImfOutputFile.h"))
         SmartPkgEnable("VRPN",      "",          ("vrpn", "quat"), ("vrpn", "quat.h", "vrpn/vrpn_Types.h"))
-        SmartPkgEnable("BULLET", "bullet", ("BulletSoftBody", "BulletDynamics", "BulletCollision", "LinearMath"), ("bullet", "bullet/btBulletDynamicsCommon.h"))
+        SmartPkgEnable("BULLET", "bullet", ("BulletSoftBody", "BulletDynamics", "BulletCollision", "LinearMath", "Bullet3Common", "Bullet3Collision", "Bullet3Dynamics", "Bullet3Geometry", "Bullet3OpenCL_clew"), ("bullet", "bullet/btBulletDynamicsCommon.h"))
         SmartPkgEnable("VORBIS",    "vorbisfile",("vorbisfile", "vorbis", "ogg"), ("ogg/ogg.h", "vorbis/vorbisfile.h"))
         SmartPkgEnable("OPUS",      "opusfile",  ("opusfile", "opus", "ogg"), ("ogg/ogg.h", "opus/opusfile.h", "opus"))
         SmartPkgEnable("JPEG",      "",          ("jpeg"), "jpeglib.h")
