diff --git a/makepanda/makepanda.py b/makepanda/makepanda.py
index 23fd57afaa..062c6d324a 100755
--- a/makepanda/makepanda.py
+++ b/makepanda/makepanda.py
@@ -88,7 +88,7 @@ PkgListSet(["PYTHON", "DIRECT",                        # Python support
   "SPEEDTREE",                                         # SpeedTree
   "ZLIB", "PNG", "JPEG", "TIFF", "OPENEXR", "SQUISH",  # 2D Formats support
   ] + MAYAVERSIONS + MAXVERSIONS + [ "FCOLLADA", "ASSIMP", "EGG", # 3D Formats support
-  "FREETYPE", "HARFBUZZ",                              # Text rendering
+  "FREETYPE", "HARFBUZZ", "BROTLI",                    # Text rendering
   "VRPN", "OPENSSL",                                   # Transport
   "FFTW",                                              # Algorithm helpers
   "ARTOOLKIT", "OPENCV", "DIRECTCAM", "VISION",        # Augmented Reality
@@ -781,27 +781,9 @@ if (COMPILER == "MSVC"):
     if (PkgSkip("DIRECTCAM")==0): LibName("DIRECTCAM", "odbccp32.lib")
     if (PkgSkip("MIMALLOC")==0): LibName("MIMALLOC", GetThirdpartyDir() + "mimalloc/lib/mimalloc-static.lib")
     if (PkgSkip("OPENSSL")==0):
-        if os.path.isfile(GetThirdpartyDir() + "openssl/lib/libpandassl.lib"):
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/libpandassl.lib")
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/libpandaeay.lib")
-        elif os.path.isfile(GetThirdpartyDir() + "openssl/lib/ssleay32.lib"):
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/libeay32.lib")
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/ssleay32.lib")
-        else:
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/libssl.lib")
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/libcrypto.lib")
-            LibName("OPENSSL", "crypt32.lib")
-            LibName("OPENSSL", "ws2_32.lib")
+        SmartPkgEnable("OPENSSL", None, ("libeay32", "ssleay32"), ("openssl/ssl.h", "openssl/crypto.h"))
     if (PkgSkip("PNG")==0):
-        if os.path.isfile(GetThirdpartyDir() + "png/lib/libpng16_static.lib"):
-            LibName("PNG", GetThirdpartyDir() + "png/lib/libpng16_static.lib")
-        else:
-            LibName("PNG", GetThirdpartyDir() + "png/lib/libpng_static.lib")
-    if (PkgSkip("TIFF")==0):
-        if os.path.isfile(GetThirdpartyDir() + "tiff/lib/libtiff.lib"):
-            LibName("TIFF", GetThirdpartyDir() + "tiff/lib/libtiff.lib")
-        else:
-            LibName("TIFF", GetThirdpartyDir() + "tiff/lib/tiff.lib")
+        SmartPkgEnable("PNG", None, ("libpng"), "png.h")
     if (PkgSkip("OPENEXR")==0):
         if os.path.isfile(GetThirdpartyDir() + "openexr/lib/OpenEXRCore-3_1.lib"):
             LibName("OPENEXR", GetThirdpartyDir() + "openexr/lib/OpenEXR-3_1.lib")
@@ -838,17 +820,21 @@ if (COMPILER == "MSVC"):
                 LibName("OPENEXR", GetThirdpartyDir() + "openexr/lib/Half" + suffix + ".lib")
         IncDirectory("OPENEXR", GetThirdpartyDir() + "openexr/include/OpenEXR")
         IncDirectory("OPENEXR", GetThirdpartyDir() + "openexr/include/Imath")
-    if (PkgSkip("JPEG")==0):     LibName("JPEG",     GetThirdpartyDir() + "jpeg/lib/jpeg-static.lib")
-    if (PkgSkip("ZLIB")==0):     LibName("ZLIB",     GetThirdpartyDir() + "zlib/lib/zlibstatic.lib")
+    if (PkgSkip("JPEG")==0):
+        SmartPkgEnable("JPEG", None, ("libjpeg"), "jpeglib.h")
+    if (PkgSkip("ZLIB")==0):
+        SmartPkgEnable("ZLIB", None, ("libz_st"), "zlib.h")
     if (PkgSkip("VRPN")==0):     LibName("VRPN",     GetThirdpartyDir() + "vrpn/lib/vrpn.lib")
     if (PkgSkip("VRPN")==0):     LibName("VRPN",     GetThirdpartyDir() + "vrpn/lib/quat.lib")
     if (PkgSkip("NVIDIACG")==0): LibName("CGGL",     GetThirdpartyDir() + "nvidiacg/lib/cgGL.lib")
     if (PkgSkip("NVIDIACG")==0): LibName("CGDX9",    GetThirdpartyDir() + "nvidiacg/lib/cgD3D9.lib")
     if (PkgSkip("NVIDIACG")==0): LibName("NVIDIACG", GetThirdpartyDir() + "nvidiacg/lib/cg.lib")
-    if (PkgSkip("FREETYPE")==0): LibName("FREETYPE", GetThirdpartyDir() + "freetype/lib/freetype.lib")
+    if (PkgSkip("FREETYPE")==0):
+        SmartPkgEnable("FREETYPE", None, ("freetype2ST"), ("freetype2", "freetype2/freetype/freetype.h"))
     if (PkgSkip("HARFBUZZ")==0):
-        LibName("HARFBUZZ", GetThirdpartyDir() + "harfbuzz/lib/harfbuzz.lib")
-        IncDirectory("HARFBUZZ", GetThirdpartyDir() + "harfbuzz/include/harfbuzz")
+        SmartPkgEnable("HARFBUZZ", None, ("libharfbuzz"), ("harfbuzz", "harfbuzz/hb-ft.h"))
+    if (PkgSkip("BROTLI")==0):
+        SmartPkgEnable("BROTLI", None, ("brotlidec-static", "brotlicommon-static"), ("brotli", "brotli/types.h"))
     if (PkgSkip("FFTW")==0):     LibName("FFTW",     GetThirdpartyDir() + "fftw/lib/fftw3.lib")
     if (PkgSkip("ARTOOLKIT")==0):LibName("ARTOOLKIT",GetThirdpartyDir() + "artoolkit/lib/libAR.lib")
     if (PkgSkip("OPENCV")==0):   LibName("OPENCV",   GetThirdpartyDir() + "opencv/lib/cv.lib")
@@ -882,10 +868,7 @@ if (COMPILER == "MSVC"):
         if (GetOptimize() <= 3):
             LibName("ROCKET", GetThirdpartyDir() + "rocket/lib/RocketDebugger.lib")
     if (PkgSkip("OPENAL")==0):
-        LibName("OPENAL", GetThirdpartyDir() + "openal/lib/OpenAL32.lib")
-        if not os.path.isfile(GetThirdpartyDir() + "openal/bin/OpenAL32.dll"):
-            # Link OpenAL Soft statically.
-            DefSymbol("OPENAL", "AL_LIBTYPE_STATIC")
+        SmartPkgEnable("OPENAL", None, ("libopenal32"), "AL/al.h")
     if (PkgSkip("ODE")==0):
         LibName("ODE",      GetThirdpartyDir() + "ode/lib/ode_single.lib")
         DefSymbol("ODE",    "dSINGLE", "")
@@ -906,11 +889,7 @@ if (COMPILER == "MSVC"):
         DefSymbol("WX",     "_UNICODE", "")
         DefSymbol("WX",     "UNICODE", "")
     if (PkgSkip("VORBIS")==0):
-        for lib in ('ogg', 'vorbis', 'vorbisfile'):
-            path = GetThirdpartyDir() + "vorbis/lib/lib{0}_static.lib".format(lib)
-            if not os.path.isfile(path):
-                path = GetThirdpartyDir() + "vorbis/lib/{0}.lib".format(lib)
-            LibName("VORBIS", path)
+        SmartPkgEnable("VORBIS", None, ("libvorbisfile_static.lib", "libvorbis_static.lib", "libogg_static.lib"), ("ogg/ogg.h", "vorbis/vorbisfile.h"))
     if (PkgSkip("OPUS")==0):
         IncDirectory("OPUS", GetThirdpartyDir() + "opus/include/opus")
         for lib in ('ogg', 'opus', 'opusfile'):
@@ -969,13 +948,7 @@ if (COMPILER == "MSVC"):
             LibName("SPEEDTREE",  "glu32.lib")
         IncDirectory("SPEEDTREE", SDK["SPEEDTREE"] + "/Include")
     if (PkgSkip("BULLET")==0):
-        suffix = '.lib'
-        if GetTargetArch() == 'x64' and os.path.isfile(GetThirdpartyDir() + "bullet/lib/BulletCollision_x64.lib"):
-            suffix = '_x64.lib'
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/LinearMath" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletCollision" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletDynamics" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletSoftBody" + suffix)
+        SmartPkgEnable("BULLET", None, ("BulletSoftBody", "BulletDynamics", "BulletCollision", "LinearMath", "Bullet3Common", "Bullet3Collision", "Bullet3Dynamics", "Bullet3Geometry", "Bullet3OpenCL_clew"), ("bullet", "bullet/btBulletDynamicsCommon.h"))
 
 if (COMPILER=="GCC"):
     PkgDisable("MIMALLOC") # no discernable benefit over glibc
@@ -1040,6 +1013,7 @@ if (COMPILER=="GCC"):
         SmartPkgEnable("FMODEX",    "",          ("fmodex"), ("fmodex", "fmodex/fmod.h"))
         SmartPkgEnable("FREETYPE",  "freetype2", ("freetype"), ("freetype2", "freetype2/freetype/freetype.h"))
         SmartPkgEnable("HARFBUZZ",  "harfbuzz",  ("harfbuzz"), ("harfbuzz", "harfbuzz/hb-ft.h"))
+        SmartPkgEnable("BROTLI",    "brotli",    ("brotlienc-static", "brotlidec-static", "brotlicommon-static"), ("brotli", "brotli/types.h"))
         SmartPkgEnable("GL",        "gl",        ("GL"), ("GL/gl.h"), framework = "OpenGL")
         SmartPkgEnable("GLES",      "glesv1_cm", ("GLESv1_CM"), ("GLES/gl.h"), framework = "OpenGLES")
         SmartPkgEnable("GLES2",     "glesv2",    ("GLESv2"), ("GLES2/gl2.h")) #framework = "OpenGLES"?
@@ -1051,7 +1025,7 @@ if (COMPILER=="GCC"):
         SmartPkgEnable("TIFF",      "libtiff-4", ("tiff"), "tiff.h")
         SmartPkgEnable("OPENEXR",   "OpenEXR",   ("IlmImf", "Imath", "Half", "Iex", "IexMath", "IlmThread"), ("OpenEXR", "Imath", "OpenEXR/ImfOutputFile.h"))
         SmartPkgEnable("VRPN",      "",          ("vrpn", "quat"), ("vrpn", "quat.h", "vrpn/vrpn_Types.h"))
-        SmartPkgEnable("BULLET", "bullet", ("BulletSoftBody", "BulletDynamics", "BulletCollision", "LinearMath"), ("bullet", "bullet/btBulletDynamicsCommon.h"))
+        SmartPkgEnable("BULLET", "bullet", ("BulletSoftBody", "BulletDynamics", "BulletCollision", "LinearMath", "Bullet3Common", "Bullet3Collision", "Bullet3Dynamics", "Bullet3Geometry", "Bullet3OpenCL_clew"), ("bullet", "bullet/btBulletDynamicsCommon.h"))
         SmartPkgEnable("VORBIS",    "vorbisfile",("vorbisfile", "vorbis", "ogg"), ("ogg/ogg.h", "vorbis/vorbisfile.h"))
         SmartPkgEnable("OPUS",      "opusfile",  ("opusfile", "opus", "ogg"), ("ogg/ogg.h", "opus/opusfile.h", "opus"))
         SmartPkgEnable("JPEG",      "",          ("jpeg"), "jpeglib.h")
@@ -1108,12 +1082,6 @@ if (COMPILER=="GCC"):
                 LibName("OPENEXR", "-Wl,--exclude-libs,libIlmThread.a")
                 LibName("OPENEXR", "-Wl,--exclude-libs,libImath.a")
 
-            if not PkgSkip("VORBIS"):
-                LibName("VORBIS", "-Wl,--exclude-libs,libogg.a")
-                LibName("VORBIS", "-Wl,--exclude-libs,libvorbis.a")
-                LibName("VORBIS", "-Wl,--exclude-libs,libvorbisenc.a")
-                LibName("VORBIS", "-Wl,--exclude-libs,libvorbisfile.a")
-
             if not PkgSkip("OPUS"):
                 LibName("OPUS", "-Wl,--exclude-libs,libogg.a")
                 LibName("OPUS", "-Wl,--exclude-libs,libopus.a")
@@ -4385,10 +4353,10 @@ if (not RUNTIME):
 #
 
 if (PkgSkip("FREETYPE")==0 and not RUNTIME):
-  OPTS=['DIR:panda/src/pnmtext', 'BUILDING:PANDA',  'FREETYPE']
+  OPTS=['DIR:panda/src/pnmtext', 'BUILDING:PANDA',  'FREETYPE', 'BROTLI']
   TargetAdd('p3pnmtext_composite1.obj', opts=OPTS, input='p3pnmtext_composite1.cxx')
 
-  OPTS=['DIR:panda/src/pnmtext', 'FREETYPE']
+  OPTS=['DIR:panda/src/pnmtext', 'FREETYPE', 'BROTLI']
   IGATEFILES=GetDirectoryContents('panda/src/pnmtext', ["*.h", "*_composite*.cxx"])
   TargetAdd('libp3pnmtext.in', opts=OPTS, input=IGATEFILES)
   TargetAdd('libp3pnmtext.in', opts=['IMOD:panda3d.core', 'ILIB:libp3pnmtext', 'SRCDIR:panda/src/pnmtext'])
@@ -4401,11 +4369,11 @@ if (not RUNTIME):
   if not PkgSkip("HARFBUZZ"):
     DefSymbol("HARFBUZZ", "HAVE_HARFBUZZ")
 
-  OPTS=['DIR:panda/src/text', 'BUILDING:PANDA', 'ZLIB',  'FREETYPE', 'HARFBUZZ']
+  OPTS=['DIR:panda/src/text', 'BUILDING:PANDA', 'ZLIB',  'FREETYPE', 'HARFBUZZ', 'BROTLI']
   TargetAdd('p3text_composite1.obj', opts=OPTS, input='p3text_composite1.cxx')
   TargetAdd('p3text_composite2.obj', opts=OPTS, input='p3text_composite2.cxx')
 
-  OPTS=['DIR:panda/src/text', 'ZLIB', 'FREETYPE']
+  OPTS=['DIR:panda/src/text', 'ZLIB', 'FREETYPE', 'BROTLI']
   IGATEFILES=GetDirectoryContents('panda/src/text', ["*.h", "*_composite*.cxx"])
   TargetAdd('libp3text.in', opts=OPTS, input=IGATEFILES)
   TargetAdd('libp3text.in', opts=['IMOD:panda3d.core', 'ILIB:libp3text', 'SRCDIR:panda/src/text'])
@@ -4544,7 +4512,7 @@ if (not RUNTIME):
 
 if (not RUNTIME):
   OPTS=['DIR:panda/metalibs/panda', 'BUILDING:PANDA', 'JPEG', 'PNG', 'HARFBUZZ',
-      'TIFF', 'OPENEXR', 'ZLIB', 'OPENSSL', 'FREETYPE', 'FFTW', 'ADVAPI', 'WINSOCK2',
+      'TIFF', 'OPENEXR', 'ZLIB', 'OPENSSL', 'FREETYPE', 'BROTLI', 'FFTW', 'ADVAPI', 'WINSOCK2',
       'SQUISH', 'NVIDIACG', 'VORBIS', 'OPUS', 'WINUSER', 'WINMM', 'WINGDI', 'IPHLPAPI',
       'SETUPAPI', 'INOTIFY']
 
@@ -4622,6 +4590,7 @@ if (not RUNTIME):
     TargetAdd('libpanda.dll', input="p3pnmtext_composite1.obj")
 
   TargetAdd('libpanda.dll', dep='dtool_have_freetype.dat')
+  TargetAdd('libpanda.dll', dep='dtool_have_brotli.dat')
   TargetAdd('libpanda.dll', opts=OPTS)
 
   PyTargetAdd('core_module.obj', input='libp3dtoolbase.in')
@@ -6285,7 +6254,7 @@ if not PkgSkip("PANDATOOL") and not PkgSkip("EGG"):
 #
 
 if not PkgSkip("FREETYPE") and not PkgSkip("PANDATOOL") and not PkgSkip("EGG"):
-  OPTS=['DIR:pandatool/src/egg-mkfont', 'DIR:pandatool/src/palettizer', 'FREETYPE']
+  OPTS=['DIR:pandatool/src/egg-mkfont', 'DIR:pandatool/src/palettizer', 'FREETYPE', 'BROTLI']
   TargetAdd('egg-mkfont_eggMakeFont.obj', opts=OPTS, input='eggMakeFont.cxx')
   TargetAdd('egg-mkfont_rangeDescription.obj', opts=OPTS, input='rangeDescription.cxx')
   TargetAdd('egg-mkfont_rangeIterator.obj', opts=OPTS, input='rangeIterator.cxx')
@@ -6294,7 +6263,7 @@ if not PkgSkip("FREETYPE") and not PkgSkip("PANDATOOL") and not PkgSkip("EGG"):
   TargetAdd('egg-mkfont.exe', input='egg-mkfont_rangeIterator.obj')
   TargetAdd('egg-mkfont.exe', input='libp3palettizer.lib')
   TargetAdd('egg-mkfont.exe', input=COMMON_EGG2X_LIBS)
-  TargetAdd('egg-mkfont.exe', opts=['ADVAPI', 'FREETYPE'])
+  TargetAdd('egg-mkfont.exe', opts=['ADVAPI', 'FREETYPE', 'BROTLI'])
 
 #
 # DIRECTORY: pandatool/src/eggcharbase/
@@ -6321,7 +6290,7 @@ if not PkgSkip("PANDATOOL") and not PkgSkip("EGG"):
   TargetAdd('egg-optchar.exe', input='egg-optchar_vertexMembership.obj')
   TargetAdd('egg-optchar.exe', input='libp3eggcharbase.lib')
   TargetAdd('egg-optchar.exe', input=COMMON_EGG2X_LIBS)
-  TargetAdd('egg-optchar.exe', opts=['ADVAPI', 'FREETYPE'])
+  TargetAdd('egg-optchar.exe', opts=['ADVAPI', 'FREETYPE', 'BROTLI'])
 
 #
 # DIRECTORY: pandatool/src/egg-palettize/
diff --git a/makepanda/makepandacore.py b/makepanda/makepandacore.py
index ad75ae6f1d..586bccfb62 100644
--- a/makepanda/makepandacore.py
+++ b/makepanda/makepandacore.py
@@ -524,6 +524,10 @@ def GetBison():
     if BISON is not None:
         return BISON
 
+    if "BISON" in os.environ:
+        BISON = os.environ["BISON"]
+        return BISON
+
     # We now require a newer version of Bison than the one we previously
     # shipped in the win-util dir.  The new version has a 'data'
     # subdirectory, so check for that.
@@ -1686,8 +1690,12 @@ def LocateLibrary(lib, lpath=[], prefer_static=False):
                 return os.path.join(dir, 'lib%s.a' % lib)
 
     for dir in lpath:
+        print(os.path.join(dir, '%s.lib' % lib))
+
         if target == 'darwin' and os.path.isfile(os.path.join(dir, 'lib%s.dylib' % lib)):
             return os.path.join(dir, 'lib%s.dylib' % lib)
+        elif target == 'windows' and os.path.isfile(os.path.join(dir, '%s.lib' % lib)):
+            return os.path.join(dir, '%s.lib' % lib)
         elif target != 'darwin' and os.path.isfile(os.path.join(dir, 'lib%s.so' % lib)):
             return os.path.join(dir, 'lib%s.so' % lib)
         elif os.path.isfile(os.path.join(dir, 'lib%s.a' % lib)):
@@ -1875,29 +1883,31 @@ def SmartPkgEnable(pkg, pkgconfig = None, libs = None, incs = None, defs = None,
             if l.startswith("lib"):
                 libname = l[3:]
 
-            if custom_loc:
-                # Try searching in the package's LibDirectories.
-                lpath = [dir for ppkg, dir in LIBDIRECTORIES if pkg == ppkg]
-                location = LocateLibrary(libname, lpath)
-                if location is not None:
-                    LibName(target_pkg, location)
-                else:
-                    have_pkg = False
-                    print(GetColor("cyan") + "Couldn't find library lib" + libname + GetColor())
+            for libname in (libname, f'lib{libname}'):
+                if custom_loc:
+                    # Try searching in the package's LibDirectories.
+                    lpath = [dir for ppkg, dir in LIBDIRECTORIES if pkg == ppkg]
+                    location = LocateLibrary(libname, lpath)
+                    if location is not None:
+                        LibName(target_pkg, location)
+                    else:
+                        have_pkg = False
+                        print(GetColor("cyan") + "Couldn't find library lib" + libname + GetColor())
+                        print(lpath)
 
-            elif SystemLibraryExists(libname):
-                # It exists in a system library directory.
-                LibName(target_pkg, "-l" + libname)
-            else:
-                # Try searching in the package's LibDirectories.
-                lpath = [dir for ppkg, dir in LIBDIRECTORIES if pkg == ppkg or ppkg == "ALWAYS"]
-                location = LocateLibrary(libname, lpath)
-                if location is not None:
+                elif SystemLibraryExists(libname):
+                    # It exists in a system library directory.
                     LibName(target_pkg, "-l" + libname)
                 else:
-                    have_pkg = False
-                    if VERBOSE or custom_loc:
-                        print(GetColor("cyan") + "Couldn't find library lib" + libname + GetColor())
+                    # Try searching in the package's LibDirectories.
+                    lpath = [dir for ppkg, dir in LIBDIRECTORIES if pkg == ppkg or ppkg == "ALWAYS"]
+                    location = LocateLibrary(libname, lpath)
+                    if location is not None:
+                        LibName(target_pkg, "-l" + libname)
+                    else:
+                        have_pkg = False
+                        if VERBOSE or custom_loc:
+                            print(GetColor("cyan") + "Couldn't find library lib" + libname + GetColor())
 
         # Determine which include directories to look in.
         incdirs = []
