diff --git a/makepanda/makepanda.py b/makepanda/makepanda.py
index 23fd57afaa..968ed7c7ef 100755
--- a/makepanda/makepanda.py
+++ b/makepanda/makepanda.py
@@ -6973,6 +6973,7 @@ if (PkgSkip("CONTRIB")==0 and not RUNTIME):
 if not PkgSkip("CONTRIB") and not PkgSkip("PYTHON") and not RUNTIME:
   OPTS=['DIR:contrib/src/rplight', 'BUILDING:RPLIGHT']
   TargetAdd('p3rplight_composite1.obj', opts=OPTS, input='p3rplight_composite1.cxx')
+  TargetAdd('libp3rplight.lib', input='p3rplight_composite1.obj')
 
   IGATEFILES=GetDirectoryContents('contrib/src/rplight', ["*.h", "*_composite*.cxx"])
   TargetAdd('libp3rplight.in', opts=OPTS, input=IGATEFILES)
