diff --git a/panda/src/bullet/bulletCharacterControllerNode.h b/panda/src/bullet/bulletCharacterControllerNode.h
index bb7457dfe0..98d0522cee 100644
--- a/panda/src/bullet/bulletCharacterControllerNode.h
+++ b/panda/src/bullet/bulletCharacterControllerNode.h
@@ -70,15 +70,15 @@ public:
 protected:
   virtual void transform_changed();
 
+  btKinematicCharacterController *_character;
+  btPairCachingGhostObject *_ghost;
+
 private:
   CPT(TransformState) _sync;
   bool _sync_disable;
 
   BulletUpAxis _up;
 
-  btKinematicCharacterController *_character;
-  btPairCachingGhostObject *_ghost;
-
   PT(BulletShape) _shape;
 
   LVector3 _linear_movement;
diff --git a/panda/src/bullet/bulletGhostNode.h b/panda/src/bullet/bulletGhostNode.h
index ddec79641a..aebce5a191 100644
--- a/panda/src/bullet/bulletGhostNode.h
+++ b/panda/src/bullet/bulletGhostNode.h
@@ -43,20 +43,20 @@ PUBLISHED:
 public:
   virtual btCollisionObject *get_object() const;
 
-  void do_sync_p2b();
-  void do_sync_b2p();
+  virtual void do_sync_p2b();
+  virtual void do_sync_b2p();
 
 protected:
   virtual void parents_changed();
   virtual void transform_changed();
 
+  btPairCachingGhostObject *_ghost;
+
 private:
   CPT(TransformState) _sync;
   bool _sync_disable;
   bool _sync_local;
 
-  btPairCachingGhostObject *_ghost;
-
   void do_transform_changed();
 
 public:
diff --git a/panda/src/bullet/bulletWorld.h b/panda/src/bullet/bulletWorld.h
index b655e0b349..bbcd8f5912 100644
--- a/panda/src/bullet/bulletWorld.h
+++ b/panda/src/bullet/bulletWorld.h
@@ -257,20 +257,6 @@ private:
     PT(CallbackObject) _filter_callback_obj;
   };
 
-  btBroadphaseInterface *_broadphase;
-  btCollisionConfiguration *_configuration;
-  btCollisionDispatcher *_dispatcher;
-  btConstraintSolver *_solver;
-  btSoftRigidDynamicsWorld *_world;
-
-  btGhostPairCallback _ghost_cb;
-
-  FilterAlgorithm _filter_algorithm;
-  btFilterCallback1 _filter_cb1;
-  btFilterCallback2 _filter_cb2;
-  btFilterCallback3 _filter_cb3;
-  btOverlapFilterCallback *_filter_cb;
-
   PT(CallbackObject) _tick_callback_obj;
   PT(CallbackObject) _contact_added_callback_obj;
 
@@ -285,6 +271,21 @@ private:
   BulletVehicles _vehicles;
   BulletConstraints _constraints;
 
+protected:
+  btBroadphaseInterface *_broadphase;
+  btCollisionConfiguration *_configuration;
+  btCollisionDispatcher *_dispatcher;
+  btConstraintSolver *_solver;
+  btSoftRigidDynamicsWorld *_world;
+
+  btGhostPairCallback _ghost_cb;
+
+  FilterAlgorithm _filter_algorithm;
+  btFilterCallback1 _filter_cb1;
+  btFilterCallback2 _filter_cb2;
+  btFilterCallback3 _filter_cb3;
+  btOverlapFilterCallback *_filter_cb;
+
 public:
   static TypeHandle get_class_type() {
     return _type_handle;
