diff --git a/panda/src/bullet/bulletCharacterControllerNode.h b/panda/src/bullet/bulletCharacterControllerNode.h
index bb7457dfe0..98d0522cee 100644
--- a/panda/src/bullet/bulletCharacterControllerNode.h
+++ b/panda/src/bullet/bulletCharacterControllerNode.h
@@ -70,15 +70,15 @@ public:
 protected:
   virtual void transform_changed();
 
+  btKinematicCharacterController *_character;
+  btPairCachingGhostObject *_ghost;
+
 private:
   CPT(TransformState) _sync;
   bool _sync_disable;
 
   BulletUpAxis _up;
 
-  btKinematicCharacterController *_character;
-  btPairCachingGhostObject *_ghost;
-
   PT(BulletShape) _shape;
 
   LVector3 _linear_movement;
