diff --git a/panda/src/bullet/bulletCharacterControllerNode.h b/panda/src/bullet/bulletCharacterControllerNode.h
index bb7457dfe0..98d0522cee 100644
--- a/panda/src/bullet/bulletCharacterControllerNode.h
+++ b/panda/src/bullet/bulletCharacterControllerNode.h
@@ -70,15 +70,15 @@ public:
 protected:
   virtual void transform_changed();
 
+  btKinematicCharacterController *_character;
+  btPairCachingGhostObject *_ghost;
+
 private:
   CPT(TransformState) _sync;
   bool _sync_disable;
 
   BulletUpAxis _up;
 
-  btKinematicCharacterController *_character;
-  btPairCachingGhostObject *_ghost;
-
   PT(BulletShape) _shape;
 
   LVector3 _linear_movement;
diff --git a/panda/src/bullet/bulletWorld.h b/panda/src/bullet/bulletWorld.h
index b655e0b349..b923f6c346 100644
--- a/panda/src/bullet/bulletWorld.h
+++ b/panda/src/bullet/bulletWorld.h
@@ -200,9 +200,6 @@ public:
   static LightMutex &get_global_lock();
 
 private:
-  void do_sync_p2b(PN_stdfloat dt, int num_substeps);
-  void do_sync_b2p();
-
   void do_attach_ghost(BulletGhostNode *node);
   void do_remove_ghost(BulletGhostNode *node);
 
@@ -257,6 +254,12 @@ private:
     PT(CallbackObject) _filter_callback_obj;
   };
 
+  PT(CallbackObject) _tick_callback_obj;
+  PT(CallbackObject) _contact_added_callback_obj;
+
+  PT(BulletDebugNode) _debug;
+
+protected:
   btBroadphaseInterface *_broadphase;
   btCollisionConfiguration *_configuration;
   btCollisionDispatcher *_dispatcher;
@@ -265,17 +268,6 @@ private:
 
   btGhostPairCallback _ghost_cb;
 
-  FilterAlgorithm _filter_algorithm;
-  btFilterCallback1 _filter_cb1;
-  btFilterCallback2 _filter_cb2;
-  btFilterCallback3 _filter_cb3;
-  btOverlapFilterCallback *_filter_cb;
-
-  PT(CallbackObject) _tick_callback_obj;
-  PT(CallbackObject) _contact_added_callback_obj;
-
-  PT(BulletDebugNode) _debug;
-
   btSoftBodyWorldInfo _info;
 
   BulletRigidBodies _bodies;
@@ -285,6 +277,15 @@ private:
   BulletVehicles _vehicles;
   BulletConstraints _constraints;
 
+  FilterAlgorithm _filter_algorithm;
+  btFilterCallback1 _filter_cb1;
+  btFilterCallback2 _filter_cb2;
+  btFilterCallback3 _filter_cb3;
+  btOverlapFilterCallback *_filter_cb;
+
+  virtual void do_sync_p2b(PN_stdfloat dt, int num_substeps);
+  virtual void do_sync_b2p();
+
 public:
   static TypeHandle get_class_type() {
     return _type_handle;
