--- panda/src/egg/eggTexture.h.orig	2018-07-23 17:57:41.896232836 +0300
+++ panda/src/egg/eggTexture.h	2019-03-30 01:18:09.020801753 +0300
@@ -60,7 +60,8 @@
     F_rgba, F_rgbm, F_rgba12, F_rgba8, F_rgba4, F_rgba5,
     F_rgb, F_rgb12, F_rgb8, F_rgb5, F_rgb332,
     F_red, F_green, F_blue, F_alpha, F_luminance,
-    F_luminance_alpha, F_luminance_alphamask
+    F_luminance_alpha, F_luminance_alphamask,
+    F_srgb, F_srgb_alpha
   };
   enum CompressionMode {
     CM_default, CM_off, CM_on,
