diff --git a/panda/src/pgraph/transformState.cxx b/panda/src/pgraph/transformState.cxx
index fc8b3de606..30d23855c2 100644
--- a/panda/src/pgraph/transformState.cxx
+++ b/panda/src/pgraph/transformState.cxx
@@ -295,6 +295,11 @@ make_pos_hpr_scale_shear(const LVecBase3 &pos, const LVecBase3 &hpr,
 CPT(TransformState) TransformState::
 make_pos_quat_scale_shear(const LVecBase3 &pos, const LQuaternion &quat,
                           const LVecBase3 &scale, const LVecBase3 &shear) {
+  nassertr(!pos.is_nan(), make_invalid());
+  nassertr(!quat.is_nan(), make_invalid());
+  nassertr(!scale.is_nan(), make_invalid());
+  nassertr(!shear.is_nan(), make_invalid());
+
   nassertr(!(pos.is_nan() || quat.is_nan() || scale.is_nan() || shear.is_nan()) , make_invalid());
   // Make a special-case check for the identity transform.
   if (pos == LVecBase3(0.0f, 0.0f, 0.0f) &&
