diff --git a/makepanda/makepanda.py b/makepanda/makepanda.py
index 42a7874821..c20f69e646 100755
--- a/makepanda/makepanda.py
+++ b/makepanda/makepanda.py
@@ -634,17 +634,10 @@ if (COMPILER == "MSVC"):
     if (PkgSkip("DIRECTCAM")==0): LibName("DIRECTCAM", "odbc32.lib")
     if (PkgSkip("DIRECTCAM")==0): LibName("DIRECTCAM", "odbccp32.lib")
     if (PkgSkip("OPENSSL")==0):
-        if os.path.isfile(GetThirdpartyDir() + "openssl/lib/libpandassl.lib"):
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/libpandassl.lib")
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/libpandaeay.lib")
-        else:
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/libeay32.lib")
-            LibName("OPENSSL", GetThirdpartyDir() + "openssl/lib/ssleay32.lib")
+        LibName("OPENSSL", SDK["PYTHON"] + "/Library/lib/libssl.lib")
+        LibName("OPENSSL", SDK["PYTHON"] + "/Library/lib/libcrypto.lib")
     if (PkgSkip("PNG")==0):
-        if os.path.isfile(GetThirdpartyDir() + "png/lib/libpng16_static.lib"):
-            LibName("PNG", GetThirdpartyDir() + "png/lib/libpng16_static.lib")
-        else:
-            LibName("PNG", GetThirdpartyDir() + "png/lib/libpng_static.lib")
+        LibName("PNG", SDK["PYTHON"] + "/Library/lib/libpng16_static.lib")
     if (PkgSkip("TIFF")==0):
         if os.path.isfile(GetThirdpartyDir() + "tiff/lib/libtiff.lib"):
             LibName("TIFF", GetThirdpartyDir() + "tiff/lib/libtiff.lib")
@@ -671,7 +664,8 @@ if (COMPILER == "MSVC"):
         IncDirectory("OPENEXR", GetThirdpartyDir() + "openexr/include/OpenEXR")
         IncDirectory("OPENEXR", GetThirdpartyDir() + "openexr/include/Imath")
     if (PkgSkip("JPEG")==0):     LibName("JPEG",     GetThirdpartyDir() + "jpeg/lib/jpeg-static.lib")
-    if (PkgSkip("ZLIB")==0):     LibName("ZLIB",     GetThirdpartyDir() + "zlib/lib/zlibstatic.lib")
+    if (PkgSkip("ZLIB")==0):
+        LibName("ZLIB", SDK["PYTHON"] + "/Library/lib/zlibstatic.lib")
     if (PkgSkip("VRPN")==0):     LibName("VRPN",     GetThirdpartyDir() + "vrpn/lib/vrpn.lib")
     if (PkgSkip("VRPN")==0):     LibName("VRPN",     GetThirdpartyDir() + "vrpn/lib/quat.lib")
     if (PkgSkip("NVIDIACG")==0): LibName("CGGL",     GetThirdpartyDir() + "nvidiacg/lib/cgGL.lib")
@@ -770,17 +764,16 @@ if (COMPILER == "MSVC"):
         IncDirectory("SPEEDTREE", SDK["SPEEDTREE"] + "/Include")
     if (PkgSkip("BULLET")==0):
         suffix = '.lib'
-        if GetTargetArch() == 'x64' and os.path.isfile(GetThirdpartyDir() + "bullet/lib/BulletCollision_x64.lib"):
-            suffix = '_x64.lib'
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/LinearMath" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletCollision" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletDynamics" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/BulletSoftBody" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3Common" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3Collision" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3Dynamics" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3Geometry" + suffix)
-        LibName("BULLET", GetThirdpartyDir() + "bullet/lib/Bullet3OpenCL_clew" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/LinearMath" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/BulletCollision" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/BulletDynamics" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/BulletSoftBody" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/Bullet3Common" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/Bullet3Collision" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/Bullet3Dynamics" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/Bullet3Geometry" + suffix)
+        LibName("BULLET", SDK["PYTHON"] + "/Library/lib/Bullet3OpenCL_clew" + suffix)
+        IncDirectory("BULLET", SDK["PYTHON"] + "/Library/include/bullet")
 
 if (COMPILER=="GCC"):
     if GetTarget() != "darwin":
