diff --git a/makepanda/makepandacore.py b/makepanda/makepandacore.py
index fea93fd43..6b9d45331 100644
--- a/makepanda/makepandacore.py
+++ b/makepanda/makepandacore.py
@@ -1873,7 +1873,8 @@ def SdkLocatePython(prefer_thirdparty_python=False):
         return
 
     if GetTarget() == 'windows':
-        SDK["PYTHON"] = GetThirdpartyBase() + "/win-python"
+        SDK["PYTHON"] = os.environ['CONDA_PREFIX'].replace('\\', '/')
+        IncDirectory("ALWAYS", SDK["PYTHON"] + "/Library/include")
         if (GetOptimize() <= 2):
             SDK["PYTHON"] += "-dbg"
         if (GetTargetArch() == 'x64' and os.path.isdir(SDK["PYTHON"] + "-x64")):
