diff --git a/panda/src/glxdisplay/glxGraphicsStateGuardian.cxx b/panda/src/glxdisplay/glxGraphicsStateGuardian.cxx
index 83ea5f2b2f..a0066a33c5 100644
--- a/panda/src/glxdisplay/glxGraphicsStateGuardian.cxx
+++ b/panda/src/glxdisplay/glxGraphicsStateGuardian.cxx
@@ -303,6 +303,8 @@ choose_pixel_format(const FrameBufferProperties &properties,
   GLXFBConfig *configs =
     _glXChooseFBConfig(_display, _screen, attrib_list, &num_configs);
 
+  bool need_alpha = properties.get_alpha_bits() != 0;
+
   if (configs != nullptr) {
     bool context_has_pbuffer, context_has_pixmap, slow;
     int quality, i;
@@ -327,6 +329,18 @@ choose_pixel_format(const FrameBufferProperties &properties,
         continue;
       }
 
+      if (need_alpha) {
+	  _visuals = _glXGetVisualFromFBConfig(_display, configs[i]);
+	  if (!visinfo) continue;
+
+	  XRenderPictFormat* pict_format = XRenderFindVisualFormat(_display, _visuals->visual);
+	  if (!pict_format) continue;
+	  if (pict_format->direct.alphaMask == 0) continue;
+
+	  XFree(_visuals);
+	  _visuals = nullptr;
+      }
+
       if (quality > best_quality) {
         best_quality = quality;
         best_result = i;
