diff --git a/panda/src/glxdisplay/glxGraphicsStateGuardian.cxx b/panda/src/glxdisplay/glxGraphicsStateGuardian.cxx
index 83ea5f2b2f..ef609925ef 100644
--- a/panda/src/glxdisplay/glxGraphicsStateGuardian.cxx
+++ b/panda/src/glxdisplay/glxGraphicsStateGuardian.cxx
@@ -16,6 +16,8 @@
 #include "config_glgsg.h"
 #include "lightReMutexHolder.h"
 
+#include <X11/extensions/Xrender.h>
+
 #include <dlfcn.h>
 
 using std::string;
@@ -303,6 +305,8 @@ choose_pixel_format(const FrameBufferProperties &properties,
   GLXFBConfig *configs =
     _glXChooseFBConfig(_display, _screen, attrib_list, &num_configs);
 
+  bool need_alpha = properties.get_alpha_bits() != 0;
+
   if (configs != nullptr) {
     bool context_has_pbuffer, context_has_pixmap, slow;
     int quality, i;
@@ -327,6 +331,18 @@ choose_pixel_format(const FrameBufferProperties &properties,
         continue;
       }
 
+      if (need_alpha) {
+	  _visuals = _glXGetVisualFromFBConfig(_display, configs[i]);
+	  if (!_visuals) continue;
+
+	  XRenderPictFormat* pict_format = XRenderFindVisualFormat(_display, _visuals->visual);
+	  if (!pict_format) continue;
+	  if (pict_format->direct.alphaMask == 0) continue;
+
+	  XFree(_visuals);
+	  _visuals = nullptr;
+      }
+
       if (quality > best_quality) {
         best_quality = quality;
         best_result = i;
