diff --git a/panda/src/net/netDatagram.h b/panda/src/net/netDatagram.h
index 7019381a36..2521a2c1c2 100644
--- a/panda/src/net/netDatagram.h
+++ b/panda/src/net/netDatagram.h
@@ -30,7 +30,7 @@
 // can't be set arbitrarily large, although there's not much point in making
 // it larger than the system MTU, which also limits the maximum size of a UDP
 // packet.
-static const int maximum_udp_datagram = 1500;
+static const int maximum_udp_datagram = 15000;
 
 /**
  * A specific kind of Datagram, especially for sending across or receiving
