diff --git a/makepanda/makepandacore.py b/makepanda/makepandacore.py
index c18d2418d0..52e4909668 100644
--- a/makepanda/makepandacore.py
+++ b/makepanda/makepandacore.py
@@ -70,6 +70,7 @@ MSVCVERSIONINFO = {
     (14,0): {"vsversion":(14,0), "vsname":"Visual Studio 2015"},
     (14,1): {"vsversion":(15,0), "vsname":"Visual Studio 2017"},
     (14,2): {"vsversion":(16,0), "vsname":"Visual Studio 2019"},
+    (14,3): {"vsversion":(17,0), "vsname":"Visual Studio 2022"},
 }
 
 ########################################################################
