diff --git a/yabee_libs/egg_writer.py b/yabee_libs/egg_writer.py
index b75597e..20a55f2 100644
--- a/yabee_libs/egg_writer.py
+++ b/yabee_libs/egg_writer.py
@@ -1650,10 +1650,18 @@ def write_out(fname, anims, from_actions, uv_img_as_tex, sep_anim, a_only,
                 fps = bpy.context.scene.render.fps / bpy.context.scene.render.fps_base
 
                 for action in bpy.data.actions:
-                    frange = action.frame_range
-                    ac = AnimCollector(obj_list, int(frange[0]), int(frange[1]),
-                                       fps, action.name, action)
-                    anim_collectors.append(ac)
+                    anim_from, anim_to = action.frame_range
+                    if action.name in ANIMATIONS:
+                        frames = ANIMATIONS[action.name]
+                        if frames[0] is not None:
+                            anim_from = frames[0]
+                        if frames[1] is not None:
+                            anim_to = frames[1]
+                        if frames[2] is not None:
+                            fps = frames[2]
+                        ac = AnimCollector(obj_list, int(anim_from), int(anim_to),
+                                           fps, action.name, action)
+                        anim_collectors.append(ac)
             else:
                 # Export animations named in ANIMATIONS dictionary.
                 for a_name, frames in ANIMATIONS.items():
